### PR TITLE
ic-keysmith: init at 1.6.0

### PR DESCRIPTION
**Pull request by colonelpanic8**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/ic-keysmith/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "dfinity";
+    repo = "keysmith";
+    rev = "v${version}";
+    sha256 = "1z0sxirk71yabgilq8v5lz4nd2bbm1xyrd5zppif8k9jqhr6v3v3";
+  };
+
+  vendorSha256 = "1p0r15ihmnmrybf12cycbav80sdj2dv2kry66f4hjfjn6k8zb0dc";
+
+  runVend = false;
```
Comment (IvarWithoutBones) [inline pkgs/tools/security/ic-keysmith/default.nix:16]: A comment explaining why these aren't being ran might be nice.

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
Just one little nit, otherwise LGTM :)

Result of `nixpkgs-review pr 129894` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ic-keysmith</li>
  </ul>
</details>

File: pkgs/tools/security/ic-keysmith/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (SuperSandro2000) [inline pkgs/tools/security/ic-keysmith/default.nix:17]:
```suggestion
```
that is the default.

File: pkgs/tools/security/ic-keysmith/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  meta = with lib; {
+    description = "Hierarchical Deterministic Key Derivation for the Internet Computer";
+    homepage = "https://github.com/dfinity/keysmith";
+    license = licenses.mit;
+    maintainers = with maintainers; [ imalison ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/tools/security/ic-keysmith/default.nix:23]:
```suggestion
```
buildGoModule defaults to a subset of unix.

**Pull request merged**